### ISSUE TITLE
Binding socket to the same port twice does not fail

### ISSUE THREAD

**Issue by Razz4780** [metalbear-co/mirrord#1310]
### Bug Description

A process with injected mirrord layer is able to bind a socket to the same port twice. Such attempt fails without mirrord with OS error `EADDRINUSE`). Minimal example:
```rust
use std::net::UdpSocket;

fn main() {
    let _socket_1 = UdpSocket::bind("0.0.0.0:5555").unwrap();
    let err = UdpSocket::bind("0.0.0.0:5555").unwrap_err();
    eprintln!("{:?}", err);
}
```
This program fails when run with mirrord:
```
thread 'main' panicked at 'called `Result::unwrap_err()` on an `Ok` value: UdpSocket { addr: 0.0.0.0:5555, fd: 12 }', src/main.rs:5:47
```

### Steps to Reproduce

1. Prepare a binary from the provided example.
2. Prepare a local cluster.
3. Deploy a `py-serv` app from the testing guide.
4. Run the binary with mirrord and the deployed app as a target.

### Your operating system and version

Linux 5.19.0-38-generic #39~22.04.1-Ubuntu x86_64

### Local process

ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=59fec39d65f671e16fbac571b1aa031829fba964, for GNU/Linux 3.2.0, with debug_info, not stripped

**Comment by aviramha** [comment]
This is because of #1124 that we decided to do as temp fix and should be resolved with #1123

**Comment by Razz4780** [comment]
> This is because of #1124 that we decided to do as temp fix and should be resolved with #1123 I think we can track it in #1123 ?

Sure, should've checked the existing issues

**Comment by aviramha** [comment]
No problem! Always good to have more details ;)
